app/Http/Controllers/AuthController.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\User;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Hash;
+
+class AuthController extends Controller
+{
+    public function store (Request $req)
+    {
+        $req->validate([
```

Comment:
Just a minor feedback: You can make use of laravel form request and make a separate request for every action that is store/update.
Commad:php artisan make:request
Link:https://laravel.com/docs/9.x/validation#creating-form-requests
LGTM 👍🏻